src/main/java/gregtech/core/network/internal/NetworkHandler.java

```diff
@@ -0,0 +1,140 @@
+package gregtech.core.network.internal;
+
+import gregtech.api.GTValues;
+import gregtech.api.network.*;
+import gregtech.api.util.GTLog;
+import io.netty.buffer.Unpooled;
+import net.minecraft.client.network.NetHandlerPlayClient;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.network.NetHandlerPlayServer;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.IThreadListener;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import net.minecraftforge.fml.common.Loader;
+import net.minecraftforge.fml.common.LoaderState;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.network.FMLEventChannel;
+import net.minecraftforge.fml.common.network.FMLNetworkEvent;
+import net.minecraftforge.fml.common.network.NetworkRegistry;
+import net.minecraftforge.fml.common.network.NetworkRegistry.TargetPoint;
+import net.minecraftforge.fml.common.network.internal.FMLProxyPacket;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+public class NetworkHandler implements INetworkHandler {
+
+    private static final NetworkHandler INSTANCE = new NetworkHandler();
+
+    private final FMLEventChannel channel;
+    private final PacketHandler packetHandler;
+
+    private NetworkHandler() {
+        channel = NetworkRegistry.INSTANCE.newEventDrivenChannel(GTValues.MODID);
+        channel.register(this);
+        packetHandler = PacketHandler.getInstance();
+    }
+
+    public static INetworkHandler getInstance() {
+        return INSTANCE;
+    }
+
+    public void registerPacket(Class<? extends IPacket> packetClass) {
+        if (Loader.instance().hasReachedState(LoaderState.INITIALIZATION)) {
+            GTLog.logger.error("Could not register packet {}, as packet registration has ended!", packetClass.getName());
+            return;
+        }
+        boolean hasServerExecutor = IServerExecutor.class.isAssignableFrom(packetClass);
+        boolean hasClientExecutor = IClientExecutor.class.isAssignableFrom(packetClass);
+
+        if (hasServerExecutor == hasClientExecutor) {
+            if (hasServerExecutor) {
+                GTLog.logger.error("Could not register packet {}, as it is both a Server and Client executor! Only one allowed. Skipping...", packetClass.getName());
+            } else {
+                GTLog.logger.error("Could not register packet {}, as it does not have an executor! Must have either IServerExecutor OR IClientExecutor. Skipping...", packetClass.getName());
+            }
+            return;
+        }
```

Comment:
It works but its quite hard to read and understand on first try.

I would simplify it to:
```suggestion
        if (hasServerExecutor && hasClientExecutor) {
            GTLog.logger.error("Could not register packet {}, as it is both a Server and Client executor! Only one allowed. Skipping...", packetClass.getName());
            return;
        } 
            
        if (!hasServerExecutor && !hasClientExecutor) {
             GTLog.logger.error("Could not register packet {}, as it does not have an executor! Must have either IServerExecutor OR IClientExecutor. Skipping...", packetClass.getName());
            return;
        }
```